src/core/BrightnessTemperature.c

```diff
@@ -161,15 +161,14 @@ void ComputeBrightnessTemperatureBox(int snapshot) //
     vel = run_globals.reion_grids.vel;
 
     // Temporary fix to the potential units issue with the velocity field
-    // Multiply by sqrt(a) to convert Gadget internal units to proper velocities.
-    // Dividing by 1000. because I believe the units are actually m/s not km/s (too large otherwise)
-    // I am just going to divide by 1000. until I recieve confirmation
+    // Multiply by 1/a to convert SWIFT internal units to proper velocities.
+    // The  velocities need to be in physical km/s units.
     for (ii = 0; ii < local_nix; ii++) {
       for (jj = 0; jj < ReionGridDim; jj++) {
         for (kk = 0; kk < ReionGridDim; kk++) {
           i_padded = grid_index(ii, jj, kk, ReionGridDim, INDEX_PADDED);
 
-          vel[i_padded] = (float)(sqrt(1. / (1. + redshift)) * vel[i_padded] / 1000.);
+          vel[i_padded] = (float)((1. + redshift) * vel[i_padded]);
```

Comment:
Just added. I also found that in read_grids.c there was not the case for the VELOCIRAPTOR_AUG so I added that one.
I was trying to test my modification but the code doesn't run when I put TreesID= 1 (GBPtrees) as I get this error:

"Failed to open file /fred/oz113/yqin/dragons/input_trees/Genesis_L210_N4320/trees/horizontal_trees_000.hdf5Task 0"

Reply:
thanks for adding the VELOCIRAPTOR_AUG case for reading the grids!  I told Valentina to switch off 21cmfast in her project, which is the reason this didn't appear in her tests.

I think this failing is because you were still using `params/simulations/Genesis_L210_N4320.par` (?) so that `SimulationDir = /fred/oz113/yqin/dragons/input_trees/Genesis_L210_N4320`. To use Tiamat trees and set TreesID=1, we need to use e.g. Tiamat.par, so that `SimulationDir  =  /fred/oz013/simulations/Tiamat`. e.g.:

![image](https://github.com/user-attachments/assets/2d4a182a-1d33-456d-a9ee-0e1ec6206200)